src/main/java/org/gridsuite/modification/server/modifications/GeneratorScaling.java

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.gridsuite.modification.server.modifications;
+
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.reporter.TypedValue;
+import com.powsybl.iidm.modification.scalable.Scalable;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Network;
+import org.gridsuite.modification.server.ModificationType;
+import org.gridsuite.modification.server.NetworkModificationException;
+import org.gridsuite.modification.server.VariationType;
+import org.gridsuite.modification.server.dto.GeneratorScalingInfos;
+import org.gridsuite.modification.server.dto.IdentifiableAttributes;
+import org.gridsuite.modification.server.dto.ScalingVariationInfos;
+import org.gridsuite.modification.server.utils.ScalingUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static org.gridsuite.modification.server.NetworkModificationException.Type.GENERATOR_SCALING_ERROR;
+
+/**
+ * @author Seddik Yengui <Seddik.yengui at rte-france.com>
+ */
+
+public class GeneratorScaling extends AbstractScaling {
+    private final boolean isIterative;
+
+    public GeneratorScaling(GeneratorScalingInfos generatorScalableInfos) {
+        super(generatorScalableInfos);
+        this.isIterative = generatorScalableInfos.isIterative();
+    }
+
+    @Override
+    public void applyStackingUpVariation(Network network,
+                                         List<IdentifiableAttributes> identifiableAttributes,
+                                         ScalingVariationInfos generatorScalingVariation) {
+        AtomicReference<Double> sum = new AtomicReference<>(0D);
+        Scalable stackingUpScalable = Scalable.stack(identifiableAttributes.stream()
+                .map(equipment -> getScalable(equipment.getId())).toArray(Scalable[]::new));
+        stackingUpScalable.scale(network, getAsked(generatorScalingVariation, sum));
```

Comment:
You should always report at least the variation asked and the variation reallly done (result of the scale method), for all kinds of variation mode ...